src/torchmetrics/classification/exact_match.py

```diff
@@ -124,6 +124,10 @@ def update(self, preds, target) -> None:
                 preds, target, self.num_classes, self.multidim_average, self.ignore_index
             )
         preds, target = _multiclass_stat_scores_format(preds, target, 1)
+
+        if self.ignore_index is not None:
+            preds[target == self.ignore_index] = self.ignore_index
```

Comment:
this logic should be moved to this function:
[_multiclass_stat_scores_format](https://github.com/Lightning-AI/metrics/blob/f075e6a28e6f1a7e095968ecf0a1da28a5c8aa3c/src/torchmetrics/functional/classification/exact_match.py#L40-L49)